src/devices/Ft232H/Ft232HDevice.cs

```diff
@@ -0,0 +1,857 @@
+                case SpiMode.Mode1:
+                    if (settings.DataFlow == DataFlow.MsbFirst)
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesInOnMinusVeClockMSBFirst;
+                    }
+                    else
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesInOnMinusVeClockLSBFirst;
+                    }
+
+                    break;
+            }
+
+            SpiChipSelectEnable((byte)settings.ChipSelectLine, true);
+            int idx = 0;
+            Span<byte> toSend = stackalloc byte[3];
+            toSend[idx++] = clock;
+            toSend[idx++] = (byte)((buffer.Length - 1) & 0xFF);
+            toSend[idx++] = (byte)((buffer.Length - 1) >> 8);
+            Write(toSend);
+            Read(buffer);
+            SpiChipSelectEnable((byte)settings.ChipSelectLine, false);
+        }
+
+        internal void SpiWriteRead(SpiConnectionSettings settings, ReadOnlySpan<byte> bufferWrite, Span<byte> bufferRead)
+        {
+            if ((bufferRead.Length > 65535) || (bufferWrite.Length > 65535))
+            {
+                throw new ArgumentException("Buffer too large, maximum size if 65535");
+            }
+
+            byte clock;
+            switch (settings.Mode)
+            {
+                default:
+                case SpiMode.Mode3:
+                case SpiMode.Mode0:
+                    if (settings.DataFlow == DataFlow.MsbFirst)
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesOutOnMinusBytesInOnPlusVeClockMSBFirst;
+                    }
+                    else
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesOutOnMinusBytesInOnPlusVeClockLSBFirst;
+                    }
+
+                    break;
+                case SpiMode.Mode2:
+                case SpiMode.Mode1:
+                    if (settings.DataFlow == DataFlow.MsbFirst)
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesOutOnPlusBytesInOnMinusVeClockMSBFirst;
+                    }
+                    else
+                    {
+                        clock = (byte)FtOpcode.ClockDataBytesOutOnPlusBytesInOnMinusVeClockLSBFirst;
+                    }
+
+                    break;
+            }
+
+            SpiChipSelectEnable((byte)settings.ChipSelectLine, true);
+            int idx = 0;
+            Span<byte> toSend = stackalloc byte[3 + bufferWrite.Length];
+            toSend[idx++] = clock;
+            toSend[idx++] = (byte)((bufferWrite.Length - 1) & 0xFF);
+            toSend[idx++] = (byte)((bufferWrite.Length - 1) >> 8);
+            bufferWrite.CopyTo(toSend.Slice(3));
+            Write(toSend);
+            Read(bufferRead);
+            SpiChipSelectEnable((byte)settings.ChipSelectLine, false);
+        }
+
+        internal void SpiChipSelectEnable(byte chipSelect, bool enable)
+        {
+            if (chipSelect < 0)
```

Comment:
chipSelect is unsigned

Reply:
right. Will adjust. Forgot to do it there. I do support SPI Device without chipselect. Great catch!